### DISCUSSION TITLE
A way to get turn time from annotation data

### DISCUSSION THREAD

**Discussion by nikita488** [Project-OSRM/osrm-backend#6804] [Q&A]
I want to recalculate remaining travel time of route, so I'm using trip service with annotations=distance,duration,speed and increase traveled time each time i reach next coord of the path (using duration array from received annotation). The problem is that annotation durations calculated without turn time in them, so sum(annotation.durations) != route.duration, so i can't easily recalculate remaining time each time I move

**Comment by danpat** [comment]
With some careful math, you can figure out the per _step_ turn durations by doing `step.duration - sum[annotations for this step]` - it's not full resolution, but it should get you closer.

**Comment by nikita488** [comment]
Thank you, that will probably be enough for my use case
